### PR TITLE
Closes #27 by allowing any custom supplier to used

### LINKED ISSUE #27: Ability to provide your own supplier to use something other than a UUID for an id

### PR DESCRIPTION
**Pull request by cyute**

### CODE REVIEW

**Comment by JonMR** [comment]
Can you update the readme as well? Even if the supplier stuff is just mentioned in the defaults section.

**Comment by cyute** [comment]
Both of those suggestions sound great, will do.

File: src/test/java/com/serviceenabled/dropwizardrequesttracker/it/CustomSupplierIT.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	}
+	
+	@Test
+	public void trackerIdComesFromCustomSupplier() throws Exception {
+		CLIENT.resource(INITIAL_URI).post(ClientResponse.class);
+
+		assertThat(MOCK_TEST.getRequestTrackerId("custom-header"), is("SUPPLY-WHATEVER-YOU-LIKE"));
+	}
+
+	private static String resourceFilePath(final String resourceClassPathLocation) {
```
Comment (cyute) [inline src/test/java/com/serviceenabled/dropwizardrequesttracker/it/CustomSupplierIT.java]: FYI: In DW `0.8.0` this method is provided in [ResourceHelpers.java](https://github.com/dropwizard/dropwizard/blob/master/dropwizard-testing/src/main/java/io/dropwizard/testing/ResourceHelpers.java).

**Comment by JonMR** [comment]
I was thinking about this one and I was trying to put myself in a user's shoes.  If I was a user, I don't have a lot of control over how my supplier is configured.  The supplier has to have a default constructor and I don't get to instantiate it.

If I have some dependencies to generate the id (db connection, [external service](https://github.com/GeneralElectric/snowizard), etc) there's no clean way for me to add those dependencies to the supplier without resorting to adding the dependencies as statics in the supplier prior to the client and servlet filter getting initialized.

Thoughts?  I might be missing something.

**Comment by cyute** [comment]
That sounds like a good point.  I imagine that will require a completely different approach.  If anyone wants to pick it up before we meet again feel free.  Otherwise, maybe we can pair on it at the next meeting.

**Comment by JonMR** [comment]
Closing in favor of #29 which @cyute, @rmuhic,  and I paired on.

**Pull request closed without merging**